xml/System.Reflection/MetadataAssemblyResolver.xml

```diff
@@ -54,11 +54,30 @@
         <Parameter Name="assemblyName" Type="System.Reflection.AssemblyName" />
       </Parameters>
       <Docs>
-        <param name="context">To be added.</param>
-        <param name="assemblyName">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="context">The metadata load context object that represents the closed universe of Type objects loaded for inspection-only purposes.</param>
+        <param name="assemblyName">The name of the assembly.</param>
+        <summary>Returns an assembly from a given assembly name.</summary>
+        <returns>The assembly for the provided <paramref name="assemblyName" />.</returns>
+        <remarks>
+          <format type="text/markdown">
+  <![CDATA[
+
+## Remarks
+
+Binding an assembly name to an assembly occurs when <xref:System.Reflection.MetadataLoadContext.LoadAssemblyByName> is called or when a Type from one assembly has a dependency on another assembly. It should use <xref:System.Reflection.MetadataLoadContext.LoadFromStream>, <xref:System.Reflection.LoadFromAssemblyPath> or <xref:System.Reflection.LoadFromByteArray> to load the requested assembly and return it.
+
+To indicate the failure to find an assembly, the handler should return `null` rather than throwing an exception. Returning `null` commits the failure so that future attempts to load that name will fail without re-invoking the handler.
+
+If the handler throws an exception, the exception will be passed through to the application that invoked the operation that triggered the binding. The MetadataLoadContext will not catch it and no binding will occur.
+
+The handler will generally not be called more than once for the same name, unless two threads race to load the same assembly. Even in that case, one result will win and be atomically bound to the name.
+
+The <xref:System.Reflection.MetadataLoadContext> intentionally performs no ref-def matching on the returned assembly as what constitutes a ref-def match is a policy. It is also the kind of arbitrary restriction that <xref:System.Reflection.MetadataLoadContext> strives to avoid.
+
+The <xref:System.Reflection.MetadataLoadContext> cannot consume assemblies from other metadata load contexts or other type providers (such as the underlying runtime's own Reflection system). If a handler returns such an assembly, the <xref:System.Reflection.MetadataLoadContext> throws a <xref:System.FileLoadException>.
```

Comment:
```suggestion
The <xref:System.Reflection.MetadataLoadContext> cannot consume assemblies from other metadata load contexts or other type providers (such as the underlying runtime's own reflection system). If a handler returns such an assembly, the <xref:System.Reflection.MetadataLoadContext> throws a <xref:System.FileNotFoundException>.
```
But types cannot throw exceptions; only type members can.

Reply:
It's supposed to be FileLoadException, I got it from here:
[src/System.Reflection.MetadataLoadContext/src/System/Reflection/MetadataAssemblyResolver.cs#L34](https://github.com/dotnet/corefx/blob/775347f642ae10badf82bcd0352c9f316093cd6f/src/System.Reflection.MetadataLoadContext/src/System/Reflection/MetadataAssemblyResolver.cs#L34)

@steveharter @GrabYourPitchforks is the exception correct? Should we reword this sentence to clarify who will throw the exception?

Reply:
Sorry, @carlossanlop. It is FileLoadException. The broken link was caused by its being in the System.IO namespace rather than System.